src/main/java/MongoDBDAO.java

```diff
@@ -1,139 +1,140 @@
-                }
-                return returnString.toString();
-            } else {
-                return "Could not find any data with the given criterias";
-            }
-        } catch (MongoException mwe) {
-            //  Block of code to handle errors
-            return mwe.getMessage();
-        }
-    }
+    public String getWeatherstationData(String stringDate , String weatherstationName , String contentType){
+        //For Production:
+        //MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection(weatherstationName);
 
-    public String findSpecFieldsFromDatabaseDATE(String stringDate){
+        //For Development:
         MongoCollection<Document> collection = MongoConnection.getInstance("AgricircleDB").getDatabase().getCollection("Weatherstation1");
 
         StringBuilder returnString = new StringBuilder();
-        ObjectId date = DateToObjectId(stringDate);
-        FindIterable<Document> iterable = collection.find(gte("_id" , date));
-        System.out.println(iterable);
-        if (iterable != null) {
-            for (Document document : iterable) {
-                returnString.append(document.toJson());
-            }
+        ObjectId date;
+        try{
+             date = DateToObjectId(stringDate);
         }
-        if(returnString.length() == 0)
+        catch(WrongDateFormatException wdfe)
+        {
+            return wdfe.getMessage();
+        }
+
+        FindIterable<Document> iterable = collection.find(gte("_id" , date)).
+                projection(fields(include(
+                        "payload_fields.avg_wind_speed",
+                        "payload_fields.solar_radiation",
+                        "payload_fields.outside_temperature",
+                        "payload_fields.outside_humidity",
+                        "payload_fields.barometer_data",
+                        "payload_fields.rain_rate"),
+                        excludeId()));
+
+        if(contentType!= null && contentType.equals("netcdf"))
         {
-            return "We couldn't find any data with the given criteria";
+            //TODO Convert JSON to netcdf and return it.
+            return "Not Yet Implemented. Use contenttype = json";
+        }
+        else
+        {
+            if (iterable != null) {
+                returnString.append("{\"dataobjects\":[");
+                for (Document document : iterable) {
+                    returnString.append(document.toJson()).append(",");
+                }
+                returnString.delete(returnString.length() - 1, returnString.length());
+                returnString.append("]}");
+            }
         }
         return returnString.toString();
     }
 
-    private static String DateToHexString(int year, int month, int day)
-    {
+    private static String DateToHexString(int year, int month, int day) throws WrongDateFormatException {
```

Comment:
What is this for? Is it neccessary? Looks wrong.

Reply:
"this" being having the method throw WrongDateFormatException?